scalameta/semanticdb-scalac/src/main/scala/scala/meta/interactive/InteractiveSemanticdb.scala

```diff
@@ -0,0 +1,112 @@
+package scala.meta.interactive
+
+import java.io.File
+import java.net.URLClassLoader
+import scala.meta.internal.semanticdb.DatabaseOps
+import scala.reflect.io.VirtualDirectory
+import scala.tools.nsc.Settings
+import scala.tools.nsc.interactive.Global
+import scala.tools.nsc.interactive.Response
+import scala.tools.nsc.reporters.StoreReporter
+import org.langmeta.semanticdb.Document
+
+object InteractiveSemanticdb {
+
+  def newCompiler(): Global =
+    newCompiler(thisClasspath, Nil)
+  def newCompiler(scalacOptions: List[String]): Global =
+    newCompiler(thisClasspath, scalacOptions)
+
+  /** Construct new presentation compiler with given classpath and scalac flags. */
+  def newCompiler(classpath: String, scalacOptions: List[String]): Global = {
```

Comment:
It would be helpful to store the classpath, compiler version, and compiler options in the document to help determine when it needs to be rebuilt.

Reply:
That is a separate concern from this small utility.